mlflow/tensorflow/__init__.py

```diff
@@ -752,6 +751,19 @@ def autolog(
 
     atexit.register(_flush_queue)
 
+    if log_model_signatures:
+        warnings.warn(
+            " The pyfunc inference behavior of TensorFlow models logged "
+            "with signatures differs from the behavior of TensorFlow "
+            "models logged without signatures. Specifically, when a "
+            "signature is present, passing a Pandas DataFrame as "
+            "input to the pyfunc `predict()` API produces an `ndarray` "
+            "(for single-output models) or a dictionary of `str -> ndarray`:"
+            " (for multi-output models). In contrast, when a signature "
+            "is *not* present, `predict()` produces"
+            " a Pandas DataFrame output in response to a Pandas DataFrame input."
+        )
```

Comment:
@bali0019 Can we move this warning to `mlflow.keras.log_model()` and `mlflow.tensorflow.log_model()`? This is a property of model logging with signatures, not just autologging.